core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -475,7 +475,7 @@ private object RTS {
                   case IO.Tags.Fail =>
                     val io = curIo.asInstanceOf[IO.Fail[E]]
 
-                    val finalizer = unwindStack
+                    val finalizer = unwindStack(true)
```

Comment:
I think it would be more precise to say, `unwindStack(!io.cause.isInterrupted)`.

That is to say, you catch errors in precisely the case that the cause is not an interruption.